handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1857,10 +1849,10 @@ private String selectApplicationProtocol(List<String> protocols,
         @Override
         public Certificate[] getPeerCertificates() throws SSLPeerUnverifiedException {
             synchronized (ReferenceCountedOpenSslEngine.this) {
-                if (peerCerts == null || peerCerts.length == 0) {
+                if (peerCerts == null || peerCerts.isEmpty()) {
                     throw new SSLPeerUnverifiedException("peer not verified");
                 }
-                return peerCerts;
+                return peerCerts.toArray(new Certificate[peerCerts.size()]);
```

Comment:
would this change require the init code to become much more complicated? I prefer to keep it if its not significantly more complicated ... better to just allocate what we need and we don't expect it to grow after its created ... seems like a good use case for a fixed size array